components/camera/videosourcewrappers.go

```diff
@@ -197,6 +197,26 @@ func (vs *videoSource) Stream(ctx context.Context, errHandlers ...gostream.Error
 	return vs.videoSource.Stream(ctx, errHandlers...)
 }
 
+// ReadImageBytes wraps ReadImage given a mimetype to encode the image as bytes data, returning
+// supplementary metadata for downstream processing.
+// TODO(hexbabe): make function private or remove altogether once the usages are limited to this file.
+func ReadImageBytes(ctx context.Context, src gostream.VideoSource, mimeType string) ([]byte, ImageMetadata, error) {
+	img, release, err := ReadImage(ctx, src)
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+	defer release()
+	imgBytes, err := rimage.EncodeImage(ctx, img, mimeType)
```

Comment:
Isn't this going to force us to do an extra `Encode`, which is expensive? Doesn't VideoSource already implement `Image` (Which has the same signature as `ReadImageBytes`)?

Why do we need this helper?

Reply:
This helper is necessary for cameras that used adhere to the `Read`/`Stream.Next` signature to convert to serve Image. The encode here isn't really an extra encode; it would've happened anyways in `server.go` before turning into a protobuf message.